template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kafka/KafkaLdapConfigProvider.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.kafka;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+
+import java.util.List;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateComponentConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.dto.LdapView;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+
+public class KafkaLdapConfigProvider implements CmTemplateComponentConfigProvider {
+
+    private static final String LDAP_AUTH_URL = "ldap.auth.url";
+
+    private static final String LDAP_AUTH_USER_DN_TEMPLATE = "ldap.auth.user.dn.template";
+
+    private static final String LDAP_AUTH_ENABLE = "ldap.auth.enable";
+
+    @Override
+    public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
+        LdapView ldapView = source.getLdapConfig().get();
+        return List.of(
+                config(LDAP_AUTH_ENABLE, "true"),
+                config(LDAP_AUTH_URL, ldapView.getConnectionURL()),
+                config(LDAP_AUTH_USER_DN_TEMPLATE, ldapView.getUserDnPattern()));
+    }
+
+    @Override
+    public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
```

Comment:
to be more flexible in the future, I'd suggest the following in the logic:

1. Don't set LDAP_AUTH_ENABLE
2. Set the other two properties only is LDAP_AUTH_ENABLE is set to true.

This way the behaviour will be controllable from the blueprint. When Kafka will use better authentication (PAM?) then source code will not have to change, only the blueprint.